t/01-test_needle.t

```diff
@@ -214,7 +228,22 @@ ok(!defined $res, "the headline is completely different");
 
 $img1   = tinycv::read($data_dir . "inst-rescuesystem-20141027.test.png");
 $needle = needle->new($data_dir . "inst-rescuesystem-20141027.json");
-$res    = $img1->search($needle);
+($res, $cand) = $img1->search($needle);
+is_deeply(
+    $cand->[0]->{area},
+    [
+        {
+            similarity => '0',
+            x          => 245,
+            w          => 312,
+            result     => 'fail',
+            y          => 219,
+            h          => 36
+
+        }
+    ],
+    'candidate total fail, but not at 0x0'
```

Comment:
Only for this very needle. I didn't want to add another test case so I picked one of the existant. But yes, I expect the needle to appear at best match. INT_MAX is a constant, true (and was chosen when we still checked the NORM, which was between 0 and 1), but it's still factor 10-100 larger than the largest needle error in the test cases. So in case the needle is huge and the  SQL_DIFF high, 0x0 becomes less of a problem - because huge needles will start close to 0x0 anyway. But the 0x0 is a big problem for other needles.